rust/chains/hyperlane-ethereum/src/trait_builder.rs

```diff
@@ -237,3 +242,39 @@ async fn build_signing_provider<M: Middleware>(
     let signing_provider = SignerMiddleware::new(provider, signer);
     Ok(signing_provider)
 }
+
+/// Wrap the provider with a gas oracle middleware.
+/// Polygon and Mumbai require using the Polygon gas oracle, see discussion here
+/// https://github.com/foundry-rs/foundry/issues/1703.
+/// Defaults to using the provider's gas oracle.
+fn wrap_with_gas_oracle<M>(
+    provider: Arc<M>,
```

Comment:
based on how `wrap_with_gas_oracle` is used, it could just take `M` and wrap it in the `Arc` in the function body